### DISCUSSION TITLE
Question - is there an event or way to reliably know when the focus will move?

### DISCUSSION THREAD

**Discussion by Techn1x** [revolist/revogrid#613] [General]
In my grid, I have an eventlistener on `afteredit` event, which I use to send data to a backend for processing and validation.
```ts
grid.addEventListener('afteredit', (event) => {
  this.pushEvent('after_edit', { data: event.detail.data, validate: true }, (reply) => grid.source = reply.data)
))
```

However, I would like to only do the validation if the focus has moved to a new row. I was hoping to be able to do something like this;

```ts
grid.addEventListener('afteredit', async (event) => {
  const focusedCell = await grid.getFocused()
  if (event.detail.y !== focusedCell.cell.y) {
    this.pushEvent('after_edit', { data: event.detail.data, validate: true }, ....)
  } else {
    this.pushEvent('after_edit', { data: event.detail.data, validate: false }, ....)
  }
))
```

But unfortunately `getFocused()` returns the same cell being edited before it has moved. And in the case of selecting an option from a dropdown, it always returns null. In both cases, I can't know if the user has "moved on" to a new row.

I then thought about using `afterfocus` event. But then I would be sending data twice in a lot of cases (for both afteredit and afterfocus), and `afterfocus` isn't called in certain cases (like when selecting option from a dropdown, or pasting data)

```ts
grid.addEventListener('afterfocus', async (event) => {
  if (event.detail.rowIndex !== lastEditedRowIndex) {
    this.pushEvent('after_edit', { data: event.detail.data, validate: true }, ....)
  }
))
```

Any ideas on how I could reliably do some validation logic, but only when the user has finished with the row they are on? It doesn't have to be focus-based, but that's all I could think of.

**Comment by m2a2x** [comment]
Hi, you could check it here, it's an angular example but logic is the same
https://codesandbox.io/p/sandbox/rg-focus-catch-angular-standalone-3rmdt3?from-embed

**Comment by Techn1x** [comment]
thanks for the link. That example seems to use the afterfocus event, so still suffers from the issues outlined with that approach (there's no dropdown there to check that issue - but I can see that pasting data does not fire the afterfocus event, so can't use it for the processing/validation)
